### ISSUE TITLE
Legacy Sensors with no "octopus" in their name

### ISSUE THREAD

**Issue by nothernman** [BottlecapDave/HomeAssistant-OctopusEnergy#986]
### Describe the bug

Hi, first of all great work. I have had issues with getting Predbat to work and it uses the your Integration. In my HA system which is on all the lastest version i have a number of active sensors of seemingly don't have the format sensor.octopus_energy_|electricity_[0-9a-z]+_[0-9a-z]+_current_rate. They have followin sensor.electricity_21j009XXXX_20000067XXXX_current_rate and sensor.electricity_21j009XXXX_200000673XXXX_previous_rate attribute.  Every other sensor/event follows the "octopus" conventation and the only issue is these legacy sensors are still there with no correctly formated naming.  This causes integration issues

### Reproduction steps

I have deleted and re-installed but, the old style sensors still exist. 

### Expected behaviour

all sensors following the same format

### Tariff Code

E-2R-OE-FIX-12M-24-08-20-H

### Integration Version

v12.2.1

### Home Assistant Version

HA CORE 2024.8.3; HA OS 13.1

### Fresh Install?

After upgrading

### Home Assistant Logs

The logs are from Predbat but, the old style sensors names are out of HA. 

### Confirmation

- [X] I confirm that I cannot find my solution within the [documentation](https://bottlecapdave.github.io/HomeAssistant-OctopusEnergy)
- [X] I confirm that I cannot find my solution within the [FAQ](https://bottlecapdave.github.io/HomeAssistant-OctopusEnergy/faq)

**Comment by BottlecapDave** [comment]
Hello. There have been naming issues in the past with sensors where the entity id was interpreted by name of the sensor. This has been resolved in later versions, but HA holds onto the original entity id. I've recently installed the integration on a few instances and seen the correct name, so it might be you were an early adopter and HA held onto the original incorrect name. You're free to rename the entity to the desired structure and your history should be preserved and prebat should pick it up.

**Comment by nothernman** [comment]
Hi,Thank youShaunOn 30 Aug 2024, at 18:12, David Kendall ***@***.***> wrote:﻿
Hello. There have been naming issues in the past with sensors where the entity id was interpreted by name of the sensor. This has been resolved in later versions, but HA holds onto the original entity id. I've recently installed the integration on a few instances and seen the correct name, so it might be you were an early adopter and HA held onto the original incorrect name. You're free to rename the entity to the desired structure and your history should be preserved and prebat should pick it up.
